litestar/contrib/sqlalchemy/dto.py

```diff
@@ -199,15 +214,32 @@ def generate_field_definitions(cls, model_type: type[DeclarativeBase]) -> Genera
                 elif isinstance(attr, str):
                     skipped_columns.add(attr)
         for key, orm_descriptor in mapper.all_orm_descriptors.items():
-            if isinstance(orm_descriptor, hybrid_property):
+            is_hybrid_property = isinstance(orm_descriptor, hybrid_property)
+            if is_hybrid_property:
                 if orm_descriptor in seen_hybrid_descriptors:
                     continue
 
-                seen_hybrid_descriptors.add(orm_descriptor)
+                seen_hybrid_descriptors.add(orm_descriptor)  # type: ignore
 
             if key in skipped_columns:
                 continue
 
+            should_skip_column = False
```

Comment:
This logic is quite convoluted. A few comments might help to explain what this does any why.